### PR TITLE
don't force create default user notification policies

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

Related to https://github.com/grafana/oncall/issues/4410

The changes in this PR are a prerequisite to https://github.com/grafana/terraform-provider-grafana/pull/1653. See the conversation [here](https://raintank-corp.slack.com/archives/C04JCU51NF8/p1719806995902499?thread_ts=1719520920.744319&cid=C04JCU51NF8) for more context on why we decided to move away from always creating default personal notification rules for users.

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] Added the relevant release notes label (see labels prefixed w/ `release:`). These labels dictate how your PR will
    show up in the autogenerated release notes.

### CODE REVIEW

File: docs/sources/oncall-api-reference/personal_notification_rules.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ The above command returns JSON structured in the following way:
...
 | `position`  | Optional | Personal notification rules execute one after another starting from `position=0`. `Position=-1` will put the escalation policy to the end of the list. A new escalation policy created with a position of an existing escalation policy will move the old one (and all following) down on the list. |
-| `type`      |   Yes    | One of: `wait`, `notify_by_slack`, `notify_by_sms`, `notify_by_phone_call`, `notify_by_telegram`, `notify_by_email`.                                                                                                                                                                                |
+| `type`      |   Yes    | One of: `wait`, `notify_by_slack`, `notify_by_sms`, `notify_by_phone_call`, `notify_by_telegram`, `notify_by_email`, `notify_by_mobile_app`, `notify_by_mobile_app_critical`.                                                                                                                                                                                |
```
Comment (joeyorlando) [inline docs/sources/oncall-api-reference/personal_notification_rules.md:38]: these two `type`s are supported by the public API, we were just missing documentation on this. Example:
```bash
data.grafana_oncall_user.joey: Reading...
data.grafana_oncall_user.joey: Read complete after 0s [id=UVNCKC46JP91X]
grafana_oncall_user_notification_rule.joey_important_step_1: Refreshing state... [id=N9SQ2K7KAK4BJ]
grafana_oncall_user_notification_rule.joey_step_1: Refreshing state... [id=NMH6B8X6CXJQK]
grafana_oncall_user_notification_rule.joey_important_step_2: Refreshing state... [id=NRVU93Z2HD4RD]
grafana_oncall_user_notification_rule.joey_step_2: Refreshing state... [id=NEM2T1S65KTSG]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # grafana_oncall_user_notification_rule.joey_important_step_1 will be updated in-place
  ~ resource "grafana_oncall_user_notification_rule" "joey_important_step_1" {
        id        = "N9SQ2K7KAK4BJ"
      ~ type      = "notify_by_mobile_app" -> "notify_by_mobile_app_critical"
        # (4 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
grafana_oncall_user_notification_rule.joey_important_step_1: Modifying... [id=N9SQ2K7KAK4BJ]
grafana_oncall_user_notification_rule.joey_important_step_1: Modifications complete after 0s [id=N9SQ2K7KAK4BJ]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

**Comment by joeyorlando** [comment]
**TODO**
- [ ] add documentation about how we (implicitly) fallback to using e-mail if a user-to-be-notified has a blank notification policy

File: docs/sources/oncall-api-reference/personal_notification_rules.md [hunk 2]
```diff
@@ -35,8 +35,8 @@ The above command returns JSON structured in the following way:
...
-| `duration`  | Optional | A time in secs when type `wait` is chosen for `type`.                                                                                                                                                                                                                                               |
+| `type`      |   Yes    | One of: `wait`, `notify_by_slack`, `notify_by_sms`, `notify_by_phone_call`, `notify_by_telegram`, `notify_by_email`, `notify_by_mobile_app`, `notify_by_mobile_app_critical`.                                                                                                                                                                                |
+| `duration`  | Optional | A time in seconds to wait (when `type=wait`). Can be one of 60, 300, 900, 1800, or 3600.                                                                                                                                                                                                                                               |
```
Comment (joeyorlando) [inline docs/sources/oncall-api-reference/personal_notification_rules.md:39]: `duration` can only be one of these values ([public api serializer](https://github.com/grafana/oncall/blob/dev/engine/apps/public_api/serializers/personal_notification_rules.py#L21)).. we were missing docs

**Review by matiasb — APPROVED** [review]
LGTM

**Pull request merged**